dask/bag/core.py

```diff
@@ -859,26 +862,113 @@ def from_filenames(filenames, chunkbytes=None, encoding=system_encoding):
 
     if chunkbytes:
         chunkbytes = int(chunkbytes)
-        taskss = [_chunk_read_file(fn, chunkbytes, encoding) for fn in full_filenames]
+        taskss = [_chunk_read_file(fn, chunkbytes, encoding, linesep)
+                  for fn in full_filenames]
         d = dict(((name, i), task)
                  for i, task in enumerate(toolz.concat(taskss)))
     else:
         extension = os.path.splitext(filenames[0])[1].strip('.')
         myopen = opens.get(extension, open)
 
-        d = dict(((name, i), (list, (decode_sequence, encoding, (myopen, fn, 'rb'))))
+        d = dict(((name, i), (list, (decode_sequence, encoding,
+                                     (pluck, 0,
+                                      (_readlines, (myopen, fn, 'rb'),
+                                       encoding, linesep)))))
                  for i, fn in enumerate(full_filenames))
 
     return Bag(d, name, len(d))
 
 
-def _chunk_read_file(filename, chunkbytes, encoding):
+def _readlines(fo, encoding, linesep, chunksize=4096):
```

Comment:
This function is complex enough to warrant a quick docstring saying what it does.